shm/Fruit.cpp

```diff
@@ -0,0 +1,9 @@
+#include "Fruit.hpp"
+
+Fruit& Fruit::operator--() {
+    if (daysToRot_ == 0) {
+        return *this;
+    }
+    --daysToRot_;
+    return *this;
+}
```

Comment:
so it could look like this 

```suggestion
    daysToRot_ == 0 ?  : --daysToRot;
    return *this;
}
```

Seems cleaner to me

Reply:
I had to create a new review for it s previous one was resolved

Reply:
or Even
```suggestion
    daysToRot_ ?  : --daysToRot;
    return *this;
}
```

Reply:
The second one not gonna work 
![image](https://user-images.githubusercontent.com/19634288/85234625-03d4ac80-b40f-11ea-91c3-cf6ab511dfb9.png)

Reply:
I preferred first one anyway :D

Reply:
maybe  `daysToRot_ ? --daysToRot_ : 1;` will work

Reply:
I've done `daysToRot_ == 0 ? /*do nothing*/ : --daysToRot_;` 
I think it's clear what's going on there.

Reply:
Are you sure you can leave empty space for option true?

Reply:
I've checked on repl.it and it seems that the compilers generates warning:

> warning: expression result unused [-Wunused-value]
>   daysToRot_ == 0 ? /*do nothing*/ : --daysToRot_;

It could be `daysToRot_ == 0 ? 0/*do nothing*/ : --daysToRot_;`  (with additional 0 between '?' and ':') to avoid warning, but the code gets dirty.

Reply:
We can leave it as is. We do not care that it's being unused, at least I don't :) It's unused by design.